### ISSUE TITLE
Add option to stack image container to print only the image ID, for scripting

### ISSUE THREAD

**Issue by nh2** [commercialhaskell/stack#2645]
stack `Version 1.2.0`.

Currently stack image container prints output like

```
➤ stack image container
Sending build context to Docker daemon 42.72 MB
Step 1 : FROM myregistry.example.com/asdf/asdf:tag-123
 ---> abc1234abc12
Step 2 : ADD ./ /
 ---> db1176411da0
Removing intermediate container eaeaeaeaeaea
Successfully built beefbeef1234
```

Could we have an option to make the output be only `beefbeef1234` so that it can be used for scripting?

**Comment by mgsloan** [comment]
Sure, that's a cool idea!

**Comment by borsboom** [comment]
This output all comes directly from `docker build`, and it's hard to get "programmer-friendly" output from the Docker CLI.  It would be ideal if Stack directly used the Docker API rather than going through the CLI.

Another thing that complicates this issue is that `stack image container` can build multiple images, so you'd need a way to know which ID goes with which image.

Another way to achieve what you want might be to add an option to `stack image container` so that you can specify the desired tag to give all images it builds.  Your script could then explicitly use the images with that tag afterward.  This is the approach I've always taken in the past rather than relying on image IDs output by the build process.

**Comment by mgsloan** [comment]
True, this means we would need to parse the output of docker.  My suggestion would be that we do the following:

`stack image container --output-sha`.  If docker exits successfully, then attempt to parse out the SHA.  If it doesn't, dump the full output and exit with an error code.
